### DISCUSSION TITLE
Command object in CSLA 5.4.2

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#2335] [Questions] [answered]
Hi All,
I am moving a very simple application from CSLA 4.xx to 5.4.2 and I noticed that ProjectExistsCommand doesn't have a constructor while ResourceExistsCommand object have a simple constructor. How is the ProjectID set in ProjectExistsCommand?
Regards

**Comment by rockfordlhotka** [comment, marked as answer]
These days, because we have the analyzer blocking the use of the `new` keyword, you generally initialize a command object via the create operation.

```c#
[Serializable]
public class MyCommand : CommandBase<MyCommand>
{
    // declare Id property here
    
    [Create]
    [RunLocal]
    private void Create(int id)
    {
        Id = id;
    }
}
```
